Skyhigh SWG init

ckaadic opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
McAfee Web Gateway is now Skyhigh Secure Webgateway. This pack includes the rebranding, a change from javascript to python and a new command.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

RotemAmit commented:
Hi @ckaadic!
Thank you again for your contribution!
After some checks we made, it has been decided to take this issue of rebranding McAfee Web Gateway into Skyhigh Secure Webgateway internally and it will be solved by the content dev team.
Therefor I'm closing this PR. Thank you again.

eyalpalo reviewed (changes requested):
Nice work! Thank you for that contribution!
Some notes:
1. In general, comments regarding a code that is shown in several methods is relevant to all (for example, the note about ElementTree).
2. test playbook need to be added. you can find some useful info here - https://xsoar.pan.dev/docs/integrations/test-playbooks.
3. let's add some unit tests to test what happens once the API does not return result as expected or exception.

ckaadic commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.py:
```diff
@@ -0,0 +1,438 @@
+        str: 'ok' if test passed, anything else will raise an exception and will fail the test.
+    """
+
+    return "ok"
```
connectivity and authentication are covered by the the with client part in main and will return indicative messages in the added try/except block in main

ckaadic commented:
@eyalpalo I changed the code following your notes and had to add 2 additional commands to create and delete lists to make the test playbook function best. However I am not sure what you want me to test in point 3, because the API only returns valid xml or 40X in other case:
>     3. let's add some unit tests to test what happens once the API does not return result as expected or exception.

- It is not possible to test invalid input, invalid session or actions on non existing content (delete non existing entry). These all return 40X status codes which raises DemistoException from the inherited Baseclient.
- It is not possible to test invalid xml response. This will always result in an error of the xml2json function.
- The only thing I can add is missing whole parts in the xml response which is no realistic scenario. I can add some of these, if this fulfills point 3.

eyalpalo reviewed (changes requested):
Hi @ckaadic. Really nice job overall!
Almost there, some last minor comments.
Also, understood your point regarding the unit tests, let's stick to the current tests.

Also, I'm asking Shirley for doc-review, after her comments lets schedule a demo

eyalpalo commented:
@ShirleyDenkberg Can you please conduct a doc review on that PR? Thanks!

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: List Description
+      type: Unknown
+  - arguments:
+    - description: List ID
```
```suggestion
    - description: List ID.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      name: list_id
+      required: true
+      default: true
+    - description: Entry Position in table
```
```suggestion
    - description: Entry position in the table.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+    name: swg-get-list-entry
+    outputs:
+    - contextPath: SWG.ListEntries.ListID
+      description: List ID of entry's list
```
```suggestion
      description: List ID of the entry's list.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: List ID of entry's list
+      type: Unknown
+    - contextPath: SWG.ListEntries.Name
+      description: Entry Name
```
```suggestion
      description: Entry name.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: Entry Name
+      type: Unknown
+    - contextPath: SWG.ListEntries.Position
+      description: Entry position in list
```
```suggestion
      description: Entry position in the list.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: Entry position in list
+      type: Unknown
+    - contextPath: SWG.ListEntries.Description
+      description: Entry Description
```
```suggestion
      description: Entry description.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: Entry Description
+      type: Unknown
+  - arguments:
+    - description: List ID
```
```suggestion
    - description: List ID.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      name: list_id
+      required: true
+      default: true
+    - description: Entry Position in table
```
```suggestion
    - description: Entry position in the table.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+    - description: Entry Position in table
+      name: entry_pos
+      required: true
+    - description: Entry description
```
```suggestion
    - description: Entry description.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: List ID of entry's list
+      type: Unknown
+    - contextPath: SWG.ListEntries.Name
+      description: Entry Name
```
```suggestion
      description: Entry name.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: Entry position in list
+      type: Unknown
+    - contextPath: SWG.ListEntries.Description
+      description: Entry Description
```
```suggestion
      description: Entry description.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: Entry Description
+      type: Unknown
+  - arguments:
+    - description: List ID
```
```suggestion
    - description: List ID.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+    name: swg-modify-list
+    outputs:
+    - contextPath: SWG.List.ID
+      description: List ID
```
```suggestion
      description: List ID.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: List ID
+      type: Unknown
+    - contextPath: SWG.List.Title
+      description: List Title
```
```suggestion
      description: List title.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: List Title
+      type: Unknown
+    - contextPath: SWG.List.Type
+      description: List Type
```
```suggestion
      description: List type.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: List Type
+      type: Unknown
+    - contextPath: SWG.List.Description
+      description: List Description
```
```suggestion
      description: List description.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: Type for the list to be created.
+      name: type
+      required: true
+    description: Create a new list
```
```suggestion
    description: Create a new list.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: List ID
+      type: Unknown
+    - contextPath: SWG.List.Title
+      description: List Title
```
```suggestion
      description: List title.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: List Title
+      type: Unknown
+    - contextPath: SWG.List.Type
+      description: List Type
```
```suggestion
      description: List type.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: List Type
+      type: Unknown
+    - contextPath: SWG.List.Description
+      description: List Description
```
```suggestion
      description: List description.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: List Description
+      type: Unknown
+  - arguments:
+    - description: List ID
```
```suggestion
    - description: List ID.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+    name: swg-delete-list
+    outputs:
+    - contextPath: SWG.List.ID
+      description: List ID
```
```suggestion
      description: List ID.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: List ID
+      type: Unknown
+    - contextPath: SWG.List.Title
+      description: List Title
```
```suggestion
      description: List title.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: List Title
+      type: Unknown
+    - contextPath: SWG.List.Type
+      description: List Type
```
```suggestion
      description: List type.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/Integrations/SkyhighSecureWebGatewayOnPrem/SkyhighSecureWebGatewayOnPrem.yml:
```diff
@@ -0,0 +1,240 @@
+      description: List Type
+      type: Unknown
+    - contextPath: SWG.List.Description
+      description: List Description
```
```suggestion
      description: List description.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/README.md:
```diff
@@ -0,0 +1,8 @@
+Note: This is a beta pack, which lets you manage Skyhigh Secure Web Gateway lists. Since the pack is beta, it might contain bugs. Updates to the pack during the beta phase might include non-backward compatible features. We appreciate your feedback on the quality and usability of the pack to help us identify issues, fix them, and continually improve.
+
+Skyhigh Secure Web Gateway (SWG) is a cloud-native web security solution that provides the an advanced layered protection from threats and data loss with integrated RBI, CASB, and DLP capabilities in the web and cloud. It enables organizations to implement a simplified SSE architecture that delivers security, scalability, and availability for a distributed and remote workforce.
```
```suggestion
Skyhigh Secure Web Gateway (SWG) is a cloud-native web security solution that provides an advanced layered protection from threats and data loss with integrated RBI, CASB, and DLP capabilities in the web and cloud. It enables organizations to implement a simplified SSE architecture that delivers security, scalability, and availability for a distributed and remote workforce.
```

ShirleyDenkberg commented on Packs/SkyhighSecureWebGateway/README.md:
```diff
@@ -0,0 +1,8 @@
+This Skyhigh Secure Web Gateway content pack contains the **Skyhigh Secure Web Gateway** integration that manages block and allow lists within Skyhigh SWG.
+
+## What does this pack do?
+* Manages the block and allow lists within Skyhihgh SWG
```
```suggestion
Manages the block and allow lists within Skyhigh SWG.
```

ShirleyDenkberg commented:
@eyalpalo Doc review completed.

eyalpalo commented:
Demo summary : @ckaadic 
1. Change the context prefix of the get available lists command to ```List```.
2. Insert the list_entries context entry under the list object.
3. When inserting a list entry, it needs to replace the old one in the context data.
4. Delete should not write to context.
5. Integration prefix - still to decide, will be updated tomorrow.

ckaadic commented:
@eyalpalo 
Adjusted the context to meet the requirements:
* Structure of SWG.List.ListEntries
* ListEnries data gets replaced when run get-list, modify-list, insert-entry, get-list-entry
* both delete commands do not write to context anymore
* adjusted testdata, test-playbook, yml and readme to meet the new structure

eyalpalo commented:
@ckaadic Looks great, lets have a 10 minute talk to see that everything is functioning as it should be and well merge it.
Also, pay attention that you missed 2 notes from Shirley in the doc review.
